packages/babel-helpers/src/helpers.js

```diff
@@ -700,3 +700,70 @@ helpers.toPropertyKey = defineHelper(`
     }
   }
 `);
+
+/**
+ * Add a helper that will throw a useful error if the transform fails to detect the class
+ * property assignment, so users know something failed.
+ */
+helpers.initializerWarningHelper = defineHelper(`
+    export default function _initializerWarningHelper(descriptor, context){
+        throw new Error(
+          'Decorating class property failed. Please ensure that ' +
+          'proposal-class-properties is enabled and set to use loose mode. ' +
+          'To use proposal-class-properties in spec mode with decorators, wait for ' +
+          'the next major version of decorators in stage 2.'
+        );
+    }
+`);
+
+/**
+ * Add a helper to call as a replacement for class property definition.
+ */
+helpers.initializerDefineProperty = defineHelper(`
+    export default function _initializerDefineProperty(target, property, descriptor, context){
+        if (!descriptor) return;
+
+        Object.defineProperty(target, property, {
+            enumerable: descriptor.enumerable,
+            configurable: descriptor.configurable,
+            writable: descriptor.writable,
+            value: descriptor.initializer ? descriptor.initializer.call(context) : void 0,
+        });
+    }
+`);
+
+/**
+ * Add a helper to take an initial descriptor, apply some decorators to it, and optionally
+ * define the property.
+ */
+helpers.applyDecoratedDescriptor = defineHelper(`
+    export default function _applyDecoratedDescriptor(target, property, decorators, descriptor, context){
+        var desc = {};
+        Object['ke' + 'ys'](descriptor).forEach(function(key){
```

Comment:
what is the problem to use `Object.keys(descriptor)` ?

Reply:
It is kept because it was in the original helper, I will check

Reply:
Because of transform-runtime, same here https://github.com/babel/babel/pull/7017/files/805c8f7cf2dc6eea9d3dcb85b55acc6a6a3334e5#diff-ba75bd4d116cc90ad8c8002e3010e821R762